idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/resolve/ResolveProviderSyntax.kt

```diff
@@ -0,0 +1,36 @@
+package arrow.meta.ide.dsl.resolve
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.ResolveScopeProvider
+import com.intellij.psi.search.GlobalSearchScope
+
+interface ResolveProviderSyntax {
+    fun IdeMetaPlugin.addResolveScopeProvider(
+        getResolveScope: (file: VirtualFile, project: Project) -> GlobalSearchScope?
+    ): ExtensionPhase =
+            extensionProvider(
+                    ResolveScopeProvider.EP_NAME,
+                    object : ResolveScopeProvider(){
+                        override fun getResolveScope(file: VirtualFile, project: Project?): GlobalSearchScope? =
+                                getResolveScope(file, project)
+                    }
+            )
+
+    fun IdeMetaPlugin.globalSearchScope(
```

Comment:
nit: 
```suggestion
    fun ResolveProviderSyntax.globalSearchScope(
```
But feel free to merge :), whenever